pkg/controller/v1beta1/inferenceservice/components/explainer.go

```diff
@@ -85,7 +85,7 @@ func (p *Explainer) Reconcile(isvc *v1beta1.InferenceService) error {
 
 	podSpec := v1.PodSpec(isvc.Spec.Explainer.PodSpec)
 	r := knative.NewKsvcReconciler(p.client, p.scheme, objectMeta, &isvc.Spec.Explainer.ComponentExtensionSpec,
-		&podSpec, isvc.Status.Components[v1beta1.ExplainerComponent])
+		&podSpec, isvc.Status.Components[v1beta1.ExplainerComponent], constants.Explainer, isvc)
```

Comment:
I think we do not need these two extra arguments, the knative service reconciler logic is not component specific.

Reply:
In the ksvc_reconciler I need to check the component type and only inject agent sidecar to predictor:
```
func addAgentAnnotations(component constants.InferenceServiceComponent, isvc *v1beta1.InferenceService, annotations map[string]string) bool {
	if component == constants.Predictor && isvc.Spec.Predictor.GetImplementation().GetStorageUri() == nil {
```

Reply:
I moved the logic that adding agent annotation to the predictor so I no longer need to pass extra variables.